### PR TITLE
If AUTOSWITCH_FILE is not a file but a virtualenv, activate it

### PR DESCRIPTION
**Pull request by afeblot**
Implement proposal https://github.com/MichaelAquilina/zsh-autoswitch-virtualenv/issues/140#issuecomment-685901973.
This is just the code change, let's see how you feel about this before looking at doc and tests.

### CODE REVIEW

File: autoswitch_virtualenv.plugin.zsh [hunk 1]
```diff
@@ -197,10 +197,13 @@ function check_venv()
                 if type "poetry" > /dev/null && _activate_poetry; then
                     return
                 fi
-            else
+            elif [[ -f "$venv_path" ]]; then
                 local switch_to="$(<"$venv_path")"
                 _maybeworkon "$(_virtual_env_dir "$switch_to")" "virtualenv"
                 return
+            elif [[ -d "$venv_path" ]] && [[ -f "$venv_path/bin/activate" ]]; then
```
Comment (MichaelAquilina) [inline autoswitch_virtualenv.plugin.zsh:204]: could you add some code comments explaining what is going on in each of these branches? (Just like you explained in the PR description)

**Review by MichaelAquilina — COMMENTED** [review]
Overall looks good :) going to be testing this locally to make sure it works as expected with the existing workflow.

Please test the use case you are explaining yourself as well as add tests to prove it works if you are happy with it :)

Docs would definitely benefit from being updated as you mentioned yourself

File: autoswitch_virtualenv.plugin.zsh (hunk 1 above)
Comment (afeblot) [reply autoswitch_virtualenv.plugin.zsh:206]: Sure, done.

**Comment by afeblot** [comment]
@MichaelAquilina , please let me know which version of the added test_check_venv.zunit test you want me to keep.
The second one looks cleaner but takes longer, as it creates a virtualenv.

**Review by MichaelAquilina — COMMENTED** [review]
I'm away from my laptop at the moment but overall the code looks good. I'll give it a last look and merge when I'm back on

**Comment by afeblot** [comment]
Before merging, one of these 2 tests should be removed, as they test the same thing. Was just asking on which one you want to keep:
* check_venv - works as intended with .venv directory available - first way
* check_venv - works as intended with .venv directory available - second way

File: README.rst [hunk 2]
```diff
@@ -56,6 +56,10 @@ same name as the current directory but can be edited if needed). There
 is then a precommand hook that looks for a ``.venv`` file and switches
 to the name specified if one is found.
 
+If ever ``.venv`` is actually not a file but a virtualenv directory
```
[the remark this answers is not in the document]
Comment (afeblot) [reply README.rst]: thanks, done.

**Comment by MichaelAquilina** [comment]
Fixes #140

**Comment by MichaelAquilina** [comment]
@afeblot would you mind uploading your GPG public key to Github so that we can get rid of the unverified tag in your commits?

**Comment by MichaelAquilina** [comment]
@afeblot please keep the second one :) While it takes longer, it looks more likely to catch future regressions which is what is important.

**Comment by MichaelAquilina** [comment]
@afeblot please make sure to squash commits once you are done. Once everything is ready I'll go ahead and merge this PR. Thanks for your contribution!

**Comment by afeblot** [comment]
@MichaelAquilina, all done, ready for merge.

**Review by MichaelAquilina — APPROVED** [review]
Nicely done, thanks for your contribution :)

**Comment by MichaelAquilina** [comment]
https://github.com/MichaelAquilina/zsh-autoswitch-virtualenv/commit/dddac00ed1d5a0586607cf3ea18f8935ad997740

**Pull request merged**